src/ess.jl

```diff
@@ -280,10 +281,13 @@ function ess_rhat(
 
         # sum correlation estimates
         pₜ = ρ_even + ρ_odd
-        sum_pₜ = pₜ
+        sum_pₜ = zero(pₜ)
 
         k = 2
         while k < maxlag
```

Comment:
We can rework this loop a little more to avoid computing the final `ρ_odd`, which is never used.